### PR TITLE
feat: view for dispute state

### PR DESCRIPTION
**Pull request by Tulun**
Basic view state for when the user is disconnected / connected when a Request is in dispute phase
Screen Shot 2022-01-21 at 12 53 39 PM
.

### CODE REVIEW

File: src/constants/blockchain.ts [hunk 1]
```diff
@@ -50,4 +52,43 @@ export const CHAINS: Record<ChainId, ChainMetadata> = {
...
+      `https://etherscan.io/tx/${txHash}`,
+    nativeCurrency: {
+      name: "Ether",
+      symbol: "ETH",
+      decimals: 18,
+    },
+  },
 };
+
+export enum RequestState {
```
[the remark this answers is not in the document]
Comment (Tulun) [reply src/constants/blockchain.ts:70]: Seems to not be exported

Namespace '"/Users/jasonkiraly/Desktop/projects/uma/optimistic-oracle-dapp/node_modules/@uma/sdk/dist/types/oracle/types/state"' has no exported member 'RequestState'.ts(2694)

File: src/constants/blockchain.ts [hunk 2]
```diff
@@ -50,4 +52,43 @@ export const CHAINS: Record<ChainId, ChainMetadata> = {
...
+  Invalid = 0, // Never requested.
+  Requested, // 1 Requested, no other actions taken.
+  Proposed, // 2 Proposed, but not expired or disputed yet.
+  Expired, // 3 Proposed, not disputed, past liveness.
+  Disputed, // 4 Disputed, but no DVM price returned yet.
+  Resolved, // 5 Disputed and DVM price is available.
+  Settled, // 6 Final price has been set in the contract (can get here from Expired or Resolved).
+}
+
+export interface IOORequest {
```
[the remark this answers is not in the document]
Comment (Tulun) [reply src/constants/blockchain.ts]: Changed!

File: src/constants/blockchain.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (daywiss) [reply src/constants/blockchain.ts:69]: Ok ill double check it, but ok for now

File: src/components/request/RequestForm.tsx [hunk 3]
```diff
@@ -46,23 +92,40 @@ const RequestForm = () => {
...
+                  ).toFormat("hh'h':mm'min' s'sec' left")})`}
+            </ParametersValue>
           </ParametersValuesWrapper>
         </RequestFormParametersWrapper>
       </RequestFormRow>
     </RequestFormWrapper>
   );
 };
 
+//       text = Duration.fromMillis(timeLeft).toFormat("hh'h':mm'min' s'sec' left");
```
Comment (daywiss) [inline src/components/request/RequestForm.tsx:129]: Log

File: src/components/request/RequestForm.tsx [hunk 4]
```diff
@@ -46,23 +92,40 @@ const RequestForm = () => {
...
+              {requestState.reward
+                ? ethers.utils.formatUnits(requestState.reward, 18)
+                : ""}
             </ParametersValue>
           </ParametersValuesWrapper>
           <ParametersValuesWrapper>
             <ParametersValueHeader>Liveness period:</ParametersValueHeader>
-            <ParametersValue>48 hours</ParametersValue>
+            <ParametersValue>
+              {requestState.customLiveness &&
```
Comment (daywiss) [inline src/components/request/RequestForm.tsx:113]: I think when this doesnt exist, it goes to oracle default liveness, but that might be something the sdk can populate

**Review by daywiss — APPROVED** [review]

File: src/components/request/RequestForm.tsx (hunk 4 above)
Comment (Tulun) [reply src/components/request/RequestForm.tsx:113]: I think all that matters is comparing the expiration time to the current time, no?

File: src/components/request/RequestForm.tsx (hunk 3 above)
Comment (Tulun) [reply src/components/request/RequestForm.tsx:129]: Removed

**Pull request merged**